tweets/2022-03-31/2022-03-31-tweets.md

```diff
@@ -0,0 +1,317 @@
+Let's dive in and see what we can do with it!
+
+https://next.rocket.modern-web.dev/docs/guides/social-media/
+
+---
+
+First up - what the hell is an Open Graph Image?
+
+It is the preview image you see when sharing websites on Facebook, Twitter, Discord, Slack, WhatsApp, Telegram and other social media.
+As it is important that links to your website stand out from the rest of the content on social media you should add one.
+
+![Open Graph images on Twitter, Slack, Discord, Telegram, Facebook, Linkedin](./wall.png)
+
+---
+
+There are many size variations but it's usually similar.
+
+➡️ Facebook: 1200 x 628
+➡️ Twitter: 1200 x 628
+➡️ LinkedIn: 1200 x 627
+➡️ ...
+
+We will multiply that value by two to have an optimal display on retina screens.
+
+Once we have an image we add it as a meta tag to our HTML.
+
+---
+
+You may thinks "what is this fuss about?" create this image once and you are done.
+
+Not so fast! Every page should have a unique image and it should reflect the content of the page.
+
+We wanted to stand out on social media so we will need to go the extra mile.
+
+---
+
+Luckily with Rocket it's straightforward to create a unique images for each page.
+
+How does it work?
+
+1. While rendering pages with an `openGraphLayout` it will create an additional `index.opengraph.html` file
+2. During the build phase it will open that file via puppeteer and take a screenshot
+3. It injects meta tags including an absolute URL to this image
+
+---
+
+Now let's setup an Open Graph image for our site.
+
+1. Add `export const openGraphLayout = data => ...` to `recursive.data.js` so it gets injected into every page
+2. On the page itself we export some data that gets passed to the layout
+3. Define the `absoluteBaseUrl` in our rocket config as it needs to be an absolute url
+
+👉 `index.rocket.js`
+
+```js
+export const openGraphTitle = 'Welcome to My Site';
+export const subTitle = 'This is my site';
+
+export default () => html`<h1>Home</h1>`;
+```
+
+👉 `recursive.data.js`
+
+```js
+export const openGraphLayout = data => html`
+  <!DOCTYPE html>
+  <html lang="en">
+    <head>
+      <meta charset="utf-8" />
+      <style>
+        body {
+          background: #ccc;
+        }
+      </style>
+    </head>
+    <body>
+      <h1>${data.openGraphTitle}</h1>
+      <p id="sub-title">${data.subTitle || ''}</p>
+    </body>
+  </html>
+`;
+```
+
+👉 `config/rocket.config.js`
+
+```js
+/** @type {import('@rocket/cli/types/main').RocketCliOptions} */
+export default {
+  absoluteBaseUrl: 'https://my-site.com/',
+};
+```
+
+---
+
+Let's take a look
+
+1. Start Rocket via `npm start`
+2. Open the browser on `http://localhost:8000/index.opengraph.html`
+3. Open Dev Tools and set width to `1200px` and height to `628px` and zoom to `50%`
+
+We can now adjust the HTML until we are satisfied with the result.
+
+![Basic Open Graph image](./01-basic.png)
+
+---
+
+One other important thing to note is that it is highly recommended to use a web font.
```

Comment:
I don't know if this tweet is 100% necessary in here. It might be good to keep as part of the guides, but in a tweet thread, it just makes it feel more like a tutorial than a showcase

Reply:
right - will tackle that separately at some point 👍